### PR TITLE
Add additional utility libraries

### PR DESCRIPTION
**Pull request by lemonmade**
This PR extracts a few more shared libraries related to the network/ TypeScript.

### CODE REVIEW

File: packages/network/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Changelog
```
Comment (nickjs) [inline packages/network/CHANGELOG.md]: Cool 👍

File: packages/network/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## Installation
+
+```bash
+$ yarn add @shopify/network
+```
+
+## Usage
+
+Just import any of the enums or function in this package to get a friendly API to common network details. The following values are exposed from this library:
```
Comment (nickjs) [inline packages/network/README.md]: `function` -> `functions`

File: packages/network/src/network.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  RequestTimeout = 408,
+  Conflict = 409,
+  Gone = 410,
+  LengthRequired = 411,
+  PreconditionFailed = 412,
+  RequestEntityTooLarge = 413,
+  RequestUriTooLong = 414,
+  UnsupportedMediaType = 415,
+  RequestedRangeNotSatisfiable = 416,
+  ExpectationFailed = 417,
```
Comment (nickjs) [inline packages/network/src/network.ts]: Missing 418 😏

File: packages/useful-types/README.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## Installation
+
+```bash
+$ yarn add @shopify/useful-types
+```
+
+## Usage
+
+Review the type definitions from this library to see what they can do for you.
```
Comment (nickjs) [inline packages/useful-types/README.md]: I would definitely prefer to see actual explanations for each type. Parsing a complex type definition is hard enough for experienced typescript users, let alone people less familiar.

File: packages/network/package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  "bugs": {
+    "url": "https://github.com/shopify/quilt/issues"
+  },
+  "homepage": "https://github.com/Shopify/quilt/blob/master/packages/network/README.md",
+  "dependencies": {},
+  "devDependencies": {
+    "typescript": "~3.0.1"
+  },
+  "files": ["dist/*"]
```
Comment (cartogram) [inline packages/network/package.json]: Nitpick: in the other package.json, this was formatted with line breaks around the square braces.

Comment (lemonmade) [reply packages/network/package.json]: This is just Yarn formatting the one I added a dependency to, I'd rather leave alone until yarn is needed and will auto-format it.

**Pull request merged**